### PR TITLE
Configure brakeman to ignore url safe preview card urls

### PR DESCRIPTION
**Pull request by mjankowski**
These values are validated on the way to being saved, and html escaped by Rails in the views.

The inclusion of the helper method here is a bit of a workaround to give brakeman a method name to ignore.

### CODE REVIEW

File: app/views/admin/trends/links/_preview_card.html.haml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
   .batch-table__row__content.pending-account
     .pending-account__header
-      = link_to preview_card.title, preview_card.url
+      = link_to preview_card.title, encode_url(preview_card.url)
```
[the remark this answers is not in the document]
Comment (mjankowski) [reply app/views/admin/trends/links/_preview_card.html.haml:7]: Hmm, yes it does. I suspect that I verified this in irb by looking at helper output but then did not consider the subsequent escaping when rendered in view.

I'll add some specs and take another shot at this one.

[the remark this answers is not in the document]
Comment (mjankowski) [reply app/views/admin/trends/links/_preview_card.html.haml]: Updated this to change the approach, including changing the PR title/description.

Added a view spec to ensure escaping actually works as intended; and then update brakeman config to ignore this specific safe path.

Also rebased against main after merge of the other final brakeman item, and included deletion of the ignore file in this PR. If for whatever reason the deletion and/or config change should be in separate PRs outside the spec addition and helper method changes, I can do that.

**Review by ClearlyClaire — APPROVED** [review]
Looks good, although I wonder if we shouldn't have an actual validation on `PreviewCard` to check that `url` is well-formed and uses the `http` or `https` scheme (the current code ensures that but the checks are scattered).

**Comment by mjankowski** [comment]
> Looks good, although I wonder if we shouldn't have an actual validation on `PreviewCard` to check that `url` is well-formed and uses the `http` or `https` scheme (the current code ensures that but the checks are scattered).

I can add some specs for see this to confirm whether it's already caught at model level, and if not add something for that. Will do in separate PR.

**Pull request merged**